File: src/test/java/fr/deroffal/aoc/Day03Test.java
package fr.deroffal.aoc;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;

class Day03Test {

	@Test
	void input_1_doit_retourner_0() {
		assertEquals(0, Day03.part1(1));
	}

	@Test
	void input_12_doit_retourner_3() {
		assertEquals(3, Day03.part1(12));
	}

	@Test
	void input_23_doit_retourner_2() {
		assertEquals(2, Day03.part1(23));
	}

	@Test
	void input_1024_doit_retourner_31() {
		assertEquals(31, Day03.part1(1024));
	}

	@Test
	void input_361527_doit_retourner_326() {
		assertEquals(326, Day03.part1(361527));
	}

}

File: kotlin/src/test/kotlin/fr/deroffal/aoc/kotlin/Day03Test.kt
package fr.deroffal.aoc.kotlin

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.CsvSource

internal class Day03Test {

    private val day03 = Day03()

    @CsvSource(
            "0,1",
            "3,12",
            "2,23",
            "31,1024"
    )
    @ParameterizedTest(name = "Manhattan distance to  {1} is {0}")
    fun `part1 matches examples`(expected: Int, input: Int) {
        assertEquals(expected, day03.solvePart1(input))
    }

    @CsvSource(
            "2,1",
            "4,2",
            "4,3",
            "5,4",
            "10,5"
    )
    @ParameterizedTest(name = "First value larger than {1} is {0}")
    fun `part2 matches the five 1st values`(expected: Int, input: Int) {
        assertEquals(expected, day03.solvePart2(input))
    }
}